docs/RELEASE.md

```diff
@@ -0,0 +1,79 @@
+# Release
+
+This document aims to describe how to perform a release of the Horreum python library.
+
+The versioning pattern should follow the [Horreum](https://github.com/Hyperfoil/Horreum) 
+versioning scheme to keep coherence among all Horreum-related projects versions.
+
+## Procedure
+
+### Tag a new version
+
+Checkout to your branch, either a `stable` (e.g., `0.12.x`) or the `main` one.
+
+```bash
+git checkout origin/main
+```
+
+Update the project version:
+
+```bash
+poetry version patch
+```
+
+This will bump your version, from `0.12-dev` to `0.12`.
+
+To double-check the version, run:
+```bash
+poetry version                                                                                                                                                      [15:55:39]
+# horreum 0.12
+```
+
+Commit the changes and tag a new version:
+```bash
+git add .
+git commit -m "Tag version 0.12"
+git tag v0.12
+```
+
+Ensure the tag is in the form of `v$(poetry version)`.
+
+Push changes and tag:
+```bash
+git push origin main
+git push origin v0.12
+```
+
+If you are releasing a new patch from a _stable_ branch all previous operations must be performed
+from _stable_ rather than from `main`.
+
+### Create stable branch
+
+> **NOTE**: If the _stable_ branch is already existing simply skip this step at all as this means 
+> you already did the following steps.
+
+To create a _stable_ branch from the `main` one, e.g., `0.12.x`, run the following commands.
+
+```bash
+git checkout origin/main
+git checkout -b 0.12.x
```

Comment:
Assuming that you have a `main` branch and a `0.12.0` branch (or, if there are no unique commits on it, the `0.12.0` "branch" could just be a tagged commit on the `main` branch), if you then want to release `0.12.1` you have a choice:  you can either add the pertinent commits on the `0.12.0` branch (creating a new branch if it was just a tag), or you can create a tag (`0.12.1`) for the commit on `main` which you want to release.

Life gets more interesting when you want to release `0.12.2`, but it boils down to the the same choices:  you can add new commits to the `0.12.1` branch (creating it if necessary, if it is just a tag) or you can tag the existing commit that you want to release on whichever branch it is on (`0.12.1` or `main`).

There is nothing which requires you to create a _branch_ in order to do a release, and there is nothing which demands that you release an update from an existing release branch.  These choices are driven by the contents of the files and changes which comprise the release (and what the ancestry of those files' commits are).

But, the real point here is, instead of checking out/creating a branch, making no changes, and pushing it up, you can do that with a single push command, which is probably much simpler.  E.g.,
```
git push origin origin/main:refs/heads/0.12.x
```
which creates the `0.12.x` branch on `origin` using the contents of `origin`'s `main` (assuming that your local tracking branch for `origin/main` is up to date with the remote `origin`).

Reply:
I see your point, but I think this is mainly releated to how anyone is used to do releases (I mean anyone has his own best practice).

> There is nothing which requires you to create a branch in order to do a release

Well that's true that to get a release/tag you don't need a branch, but the purpose of having a release branch like `0.12.x` is to have a branch where we could push/backport bug fixing (furthermore this is made easier thanks to the automated backporting process).